engine.io-server/src/main/java/io/socket/engineio/server/EngineIoSocket.java

```diff
@@ -149,6 +149,11 @@ void upgrade(final Transport transport) {
                     add(replyPacket);
                 }});
 
+                final Packet<String> noopPacket = new Packet<>(Packet.NOOP);
+                mTransport.send(new ArrayList<Packet>(){{
```

Comment:
Check if the transport is writeable using `mTransport.isWritable()` before attempting to send a NOOP.

Reply:
Also, move the `new ArrayList<Packet>(...)` to a static context to reduce the creation of instances.